spring-rabbit/src/test/java/org/springframework/amqp/rabbit/AsyncRabbitTemplateTests.java

```diff
@@ -394,6 +396,29 @@ public void testStopCancelled() throws Exception {
 		assertThat(callback.result).isNull();
 	}
 
+	@Test
+	@DirtiesContext
+	public void testConversionException() throws InterruptedException {
+		this.asyncTemplate.getRabbitTemplate().setMessageConverter(new SimpleMessageConverter() {
+			@Override
+			public Object fromMessage(Message message) throws MessageConversionException {
+				throw new MessageConversionException("Failed to convert message");
+			}
+		});
+
+		RabbitConverterFuture<String> replyFuture = this.asyncTemplate.convertSendAndReceive("conversionException");
+
+		final CountDownLatch cdl = new CountDownLatch(1);
+		final AtomicReference<Object> resultRef = new AtomicReference<>();
+        replyFuture.whenComplete((result, ex) -> {
+			resultRef.set(result);
+			cdl.countDown();
+		});
+        assertThat(cdl.await(10, TimeUnit.SECONDS)).isTrue();
+        assertThat(replyFuture).isCompletedExceptionally();
+        assertThat(resultRef.get()).isNull();
```

Comment:
But this line has to be removed then 😄

Reply:
sure :) 

Do you want me to align other tests to use this `failsWithin`

```
void testReturn() {
...
		assertThat(future)
				.as("Expected exception")
				.failsWithin(Duration.ofSeconds(10))
				.withThrowableThat()
				.withCauseInstanceOf(AmqpMessageReturnedException.class)
				.extracting("routingKey")
				.isEqualTo(this.requests.getName() + "x");
```

instead 
```
try {
			future.get(10, TimeUnit.SECONDS);
			fail("Expected exception");
		}
		catch (ExecutionException e) {
			assertThat(e.getCause()).isInstanceOf(AmqpMessageReturnedException.class);
			assertThat(((AmqpMessageReturnedException) e.getCause()).getRoutingKey()).isEqualTo(this.requests.getName() + "x");
		}
```

Reply:
I would appreciate.
Thanks.
Those `try..catch` in tests is on my TODO list in favor of `assertThat()` when I touch these classes.
Although still would be with some code.
But now I have learned about `failsWithin()` which looks much cleaner. 😄